### PR TITLE
added go vet shadow testing

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@liggitt  This adds `go tool vet -shadow`

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@liggitt rebased & updated

**Comment by stevekuznetsov** [comment]
@liggitt rebased & updated

**Comment by stevekuznetsov** [comment]
@liggitt rebased & updated

**Comment by stevekuznetsov** [comment]
I can't reproduce this locally (from Travis):
```
hack/verify-govet.sh
pkg/generate/app/app_test.go:317: ExampleGenerateSimpleDockerApp refers to unknown identifier: GenerateSimpleDockerApp
```

**Comment by stevekuznetsov** [comment]
Blocked on #4889

**Comment by ironcladlou** [comment]
@stevekuznetsov I'll review this as soon as you rebase

**Comment by stevekuznetsov** [comment]
@ironcladlou rebased & updated

File: pkg/cmd/cli/cmd/edit.go [hunk 1]
```diff
@@ -417,7 +417,7 @@ func applyPatch(delta *jsonmerge.Delta, info *resource.Info, version string) err
 // their updates were preserved.
 func preservedFile(err error, path string, out io.Writer) error {
 	if len(path) > 0 {
-		if _, err := os.Stat(path); !os.IsNotExist(err) {
+		if _, err = os.Stat(path); !os.IsNotExist(err) {
```
Comment (ironcladlou) [inline pkg/cmd/cli/cmd/edit.go]: I feel like this should result in a name change to `statErr` or something to be consistent with shadowing checks of other non-argument locals. @stevekuznetsov @liggitt thoughts?

File: test/integration/dockerregistryclient_test.go [hunk 2]
```diff
@@ -16,9 +16,9 @@ func TestRegistryClientConnect(t *testing.T) {
 		t.Fatal(err)
 	}
 	for _, s := range []string{"index.docker.io", "https://docker.io", "https://index.docker.io"} {
-		otherConn, err := c.Connect(s, false)
-		if err != nil {
-			t.Errorf("%s: can't connect: %v", s, err)
+		otherConn, connErr := c.Connect(s, false, true)
```
Comment (ironcladlou) [inline test/integration/dockerregistryclient_test.go]: I think I'm misunderstanding how this is compiling with that third `true` argument to `Connect`: https://github.com/openshift/origin/blob/master/pkg/dockerregistry/client.go#L39

Comment (stevekuznetsov) [inline test/integration/dockerregistryclient_test.go]: I think this is a rebase gaffe

File: pkg/cmd/cli/cmd/edit.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/cli/cmd/edit.go]: this sort of thing is why this PR should be reviewed slowly and carefully... I'd need to dig into what the expected behavior is

Comment (stevekuznetsov) [inline pkg/cmd/cli/cmd/edit.go]: @liggitt It isn't as scary as it looks. The code originally over-wrote the value of `err` but made it look like it didn't with `:=`. Either we change to `=` and make it obvious that this is happening or change the error name to make it not happen. I'll take the latter path as it seems safer.

**Comment by stevekuznetsov** [comment]
@liggitt @ironcladlou refactored to never result in over-writing existing values.

**Comment by smarterclayton** [comment]
This has been upstreamed now, make the change there.  We'll drop our
version soon.

**Comment by ironcladlou** [comment]
This LGTM

**Comment by danmcp** [comment]
@stevekuznetsov Still valid?  Needs rebase.

**Comment by stevekuznetsov** [comment]
We have larger issues with vet right now. What this really needs is review, whenever Jordan's pipeline clears up.

**Comment by stevekuznetsov** [comment]
Closing this to favor smaller, more focused `shadow` fixes by package so the reviewer doesn't dread this PR.

**Pull request closed without merging**